src/webgl/p5.RendererGL.js

```diff
@@ -996,75 +996,33 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
   strokeJoin(join) {
     this.curStrokeJoin = join;
   }
-
-  getFilterGraphicsLayer() {
-    // Lazily initialize the filter graphics layer. We only do this on demand
-    // because the graphics layer itself has a p5.RendererGL, which would then
-    // try to make its own filter layer, infinitely looping.
+  getFilterLayer() {
     if (!this.filterGraphicsLayer) {
-      // the real _pInst is buried when this is a secondary p5.Graphics
-
-      const pInst =
-        this._pInst instanceof p5.Graphics ? this._pInst._pInst : this._pInst;
-
-      // create secondary layer
-      this.filterGraphicsLayer =
-        new p5.Graphics(
-          this.width,
-          this.height,
-          constants.WEBGL,
-          pInst
-        );
-      // geometries/borders on this layer should always be invisible
-      this.filterGraphicsLayer.noStroke();
-    }
-    if (
-      this.filterGraphicsLayer.width !== this.width ||
-      this.filterGraphicsLayer.height !== this.height
-    ) {
-      // Resize the graphics layer
-      this.filterGraphicsLayer.resizeCanvas(this.width, this.height);
-    }
-    if (
-      this.filterGraphicsLayer.pixelDensity() !== this._pInst.pixelDensity()
-    ) {
-      this.filterGraphicsLayer.pixelDensity(this._pInst.pixelDensity());
+      this.filterGraphicsLayer = this._pInst.createFramebuffer();
     }
     return this.filterGraphicsLayer;
   }
-
-  getFilterGraphicsLayerTemp() {
-    // two-pass blur filter needs another graphics layer
+  getFilterLayerTemp() {
     if (!this.filterGraphicsLayerTemp) {
-      const pInst = this._pInst instanceof p5.Graphics ?
-        this._pInst._pInst : this._pInst;
-      // create secondary layer
-      this.filterGraphicsLayerTemp =
-        new p5.Graphics(
-          this.width,
-          this.height,
-          constants.WEBGL,
-          pInst
-        );
-      this.filterGraphicsLayerTemp.noStroke();
+      this.filterGraphicsLayerTemp = this._pInst.createFramebuffer();
     }
+    return this.filterGraphicsLayerTemp;
+  }
+  matchSize(fboToMatch, target) {
     if (
-      this.filterGraphicsLayerTemp.width !== this.width ||
-      this.filterGraphicsLayerTemp.height !== this.height
+      fboToMatch.width !== target.width ||
+      fboToMatch.height !== target.height
     ) {
-      // Resize the graphics layer
-      this.filterGraphicsLayerTemp.resizeCanvas(this.width, this.height);
+      fboToMatch.resizeCanvas(target.width, target.height);
```

Comment:
I believe this needs to just be `resize` instead of `resizeCanvas`, this currently throws errors: https://editor.p5js.org/davepagurek/sketches/jOYUVNMXB

Reply:
Okay @davepagurek

Reply:
![no errors now](https://github.com/processing/p5.js/assets/127239756/ace402fe-3f6d-4a51-9c00-3804a77be73f)

No errors now